src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -150,7 +150,7 @@ private void start() {
       List<ConnectedComponent> connectedComponents = RcaUtil.getAnalysisGraphComponents(rcaConf);
 
       // Mute the rca nodes after the graph creation and before the scheduler start
-      readAndUpdateMutesRcas();
+      readAndUpdateMutesRcasDuringStart();
```

Comment:
There is an edge case where both `readAndUpdateMutesRcas` and `readAndUpdateMutesRcasDuringStart` can try to update the muted Rca list back to back. So we would try to read in `rca.conf` twice. 

This will happen when rca was turned on and then off. Then between the two muted rcas update checks interval, some does both the things - adds a new muted rca and then turns on the rca. 

`readAndUpdateMutesRcasDuringStart` should ever be read at the start of the process itself. That's the time when RCA graph is not constructed and we cannot validate the new new muted RCAs. For any other updated to the muted list, the periodic rca.conf update checker will take care of it.

Reply:
Agreed, Have also added a comment explaining this.